### PR TITLE
Preserve inbox membershipType on metaReceivedError

### PR DESCRIPTION
**Pull request by buoyad**
Fixes the bug for reset users where implicit team chats can get in a 'waiting for rekey' state rather than 'waiting for someone to let you in'.

r? @keybase/react-hackers

### CODE REVIEW

File: shared/reducers/chat2.js [hunk 1]
```diff
@@ -81,6 +81,9 @@ const metaMapReducer = (metaMap, action) => {
               conversationIDKey,
               Constants.makeConversationMeta({
                 conversationIDKey,
```
Comment (chrisnojima) [inline shared/reducers/chat2.js]: we should just consume the entire InboxUIItemError and not just the membershipType

Comment (buoyad) [reply shared/reducers/chat2.js]: added a helper to eat the InboxItem, PTAL

File: shared/constants/chat2/meta.js [hunk 2]
```diff
@@ -80,6 +80,57 @@ export const unverifiedInboxUIItemToConversationMeta = (
   })
 }
 
+export const metaReceivedErrorToConversationMeta = (e: RPCChatTypes.InboxUIItemError, username: ?string) => {
```
Comment (chrisnojima) [inline shared/constants/chat2/meta.js]: we have a helper to consume the unverified item, why not just call that

Comment (buoyad) [reply shared/constants/chat2/meta.js]: The reducer was pulling `rekeyers` out of the error which the existing helper (`unverifiedInboxUIItemToConversationMeta`) doesn't do, so I was hesitant to toss that data. I tried using `unverifiedInboxUIItemToConversationMeta` too and it seemed to work, but I didn't test it extensively

**Pull request merged**